src/nvim/eval/window.c

```diff
@@ -732,7 +701,27 @@ void f_win_splitmove(typval_T *argvars, typval_T *rettv, EvalFuncData fptr)
     size = (int)tv_dict_get_number(d, "size");
   }
 
-  win_move_into_split(wp, targetwin, size, flags);
+  // Check if we can split the target before we bother switching windows.
+  if (is_aucmd_win(wp) || text_or_buf_locked() || check_split_disallowed(targetwin) == FAIL) {
```

Comment:
`check_split_disallowed(targetwin)` ought to be `check_split_disallowed(wp)` instead, but I'm not convinced checking `b_split_locked` is needed when split-moving anyway; we're not producing new views into a closing buffer by doing it.

I'll probably end up addressing this and others sometime after vim/vim#14109.